persistence/remote/src/main/java/org/infinispan/persistence/remote/internal/GlobalRemoteContainersImpl.java

```diff
@@ -29,27 +33,57 @@ public class GlobalRemoteContainersImpl implements GlobalRemoteContainers {
    @Inject
    BlockingManager blockingManager;
 
-   private final Map<String, CompletionStage<RemoteCacheManager>> remoteContainers = new HashMap<>();
+   private final Map<String, CompletionStage<RemoteCacheManager>> remoteContainers = new ConcurrentHashMap<>();
 
    @Start
    public void start() {
       RemoteContainersConfiguration configuration = globalConfiguration.module(RemoteContainersConfiguration.class);
       if (configuration != null) {
          for (Map.Entry<String, RemoteContainerConfiguration> e : configuration.configurations().entrySet()) {
-            ConfigurationBuilder builder = new ConfigurationBuilder();
-            builder.uri(e.getValue().uri()).transportFactory(configuration.transportFactory());
-            remoteContainers.put(e.getKey(), blockingManager.supplyBlocking(() -> new RemoteCacheManager(builder.build()), "RemoteContainer-create"));
+            remoteContainers.put(e.getKey(), CompletableFutures.completedNull());
          }
       }
    }
 
    @Stop
    public void stop() {
-      remoteContainers.values().forEach(c -> c.thenAccept(RemoteCacheManager::stop));
+      remoteContainers.values().forEach(c -> c.thenAccept(this::stop));
```

Comment:
If we remove the entries from the container and then invoke stop then this component can be restarted if necessary.

Reply:
Updated. After the close, there is still an entry on the map, but the CF returns null. If the container is requested again, then a new RCM is created.